### PR TITLE
added support for sunton 8048s070

### PR DESCRIPTION
**Pull request by SeanTheITGuy**
7" ESP32S3 Tablet. 16MB Flash, 8MB PSRAM. RGB565 Display, touch and SD card.

### CODE REVIEW

**Review by RetiredWizard — COMMENTED** [review]
I don't have the hardware but did look over the schematics.

**Comment by RetiredWizard** [comment]
ugh, I don't have this review process down yet. I lost my comments..... :frowning: I'll post them again :/

File: ports/espressif/boards/sunton_esp32_8048S070/pins.c [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    { MP_ROM_QSTR(MP_QSTR_IO19), MP_ROM_PTR(&pin_GPIO19) }, // P2, External SPI plug
+    { MP_ROM_QSTR(MP_QSTR_IO11), MP_ROM_PTR(&pin_GPIO11) },
+    { MP_ROM_QSTR(MP_QSTR_IO12), MP_ROM_PTR(&pin_GPIO12) },
+    { MP_ROM_QSTR(MP_QSTR_IO13), MP_ROM_PTR(&pin_GPIO13) },
+
+    { MP_ROM_QSTR(MP_QSTR_IO20), MP_ROM_PTR(&pin_GPIO20) }, // P3 has 19&20 for I2C bus, plus 17&18
+    { MP_ROM_QSTR(MP_QSTR_IO17), MP_ROM_PTR(&pin_GPIO17) }, // P4 & P5 are both 17,18,3.3v,G
+    { MP_ROM_QSTR(MP_QSTR_IO18), MP_ROM_PTR(&pin_GPIO18) },
+
+    // UART pins
```
Comment (RetiredWizard) [inline ports/espressif/boards/sunton_esp32_8048S070/pins.c:95]: // I2S/Speaker (P6) pins
    { MP_ROM_QSTR(MP_QSTR_I2S_BIT_CLOCK), MP_ROM_PTR(&pin_GPIO0) },
    { MP_ROM_QSTR(MP_QSTR_I2S_WORD_SELECT), MP_ROM_PTR(&pin_GPIO18) },
    { MP_ROM_QSTR(MP_QSTR_I2S_DATA), MP_ROM_PTR(&pin_GPIO17) },

It looks like there is I2S audio support for connector P6 on thses pins.

File: ports/espressif/boards/sunton_esp32_8048S070/pins.c [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    { MP_ROM_QSTR(MP_QSTR_TX), MP_ROM_PTR(&pin_GPIO43) },
+    { MP_ROM_QSTR(MP_QSTR_RX), MP_ROM_PTR(&pin_GPIO44) },
+
+    // I2C
+    { MP_ROM_QSTR(MP_QSTR_SDA), MP_ROM_PTR(&pin_GPIO19) },
+    { MP_ROM_QSTR(MP_QSTR_SCL), MP_ROM_PTR(&pin_GPIO20) },
```
Comment (RetiredWizard) [inline ports/espressif/boards/sunton_esp32_8048S070/pins.c:101]: I believe GPIO 18 and 19 are used by ESPRESSIF chips for USB D+/D- and CircuitPython doesn't share them very well (you can use them with Arduino code). I believe the Resistive Touch version of this board will have problems using the Touch Screen in CircuitPython. Do you think 17 & 18 would be a better choice for the board's generic I2C bus and pins 19 & 20 would be better named TOUCH_SCL & TOUCH_SDA? I know the touch I2C pins are broken out on P3 but given the USB conflict and that P4 and P5 look to me like they're designed for I2C bus usage as well they might be work out better.

**Review by RetiredWizard — CHANGES_REQUESTED** [review]
Hopefully I don't loose them this time :grin:

File: ports/espressif/boards/sunton_esp32_8048S070/pins.c (hunk 2 above)
Comment (SeanTheITGuy) [reply ports/espressif/boards/sunton_esp32_8048S070/pins.c:101]: I have added definitions for those pins for i2s usage.  It doesn't appear as if the i2c touch sensor (the capacitive one, which I have) has any issue with utilizing pins 19/20.  I agree that it looks like 17/18 would be good candidates for i2c, but attempting to use them results in an in use error, I'm guessing due to the i2s amplifier keeping them active.

Comment (SeanTheITGuy) [reply ports/espressif/boards/sunton_esp32_8048S070/pins.c:101]: ugh..  I've messed up the PR..

Comment (RetiredWizard) [reply ports/espressif/boards/sunton_esp32_8048S070/pins.c:101]: Yea the USB conflict probably only affects the resisitve touch screen. The I2S conflict is a good guess for the the "use error" but I'd also be surprised if attempting to use the I2C bus over pins 19&20 worked, I'd expect some sort of resource in use error from that as well.

Comment (SeanTheITGuy) [reply ports/espressif/boards/sunton_esp32_8048S070/pins.c:101]: the resistive is spi, the capacitive (the one I have) uses i2c on 19/20 and it works as far as I can tell.  touches register, no errors..

Comment (RetiredWizard) [reply ports/espressif/boards/sunton_esp32_8048S070/pins.c:101]: hmm, have you tested the touch detection in CircuitPython? I'm really shocked that it works.

Comment (SeanTheITGuy) [reply ports/espressif/boards/sunton_esp32_8048S070/pins.c:101]: Yes.

**Comment by DavePutz** [comment]
I tested display, wifi, web workflow, and sd card. All looks good to me.

**Comment by SeanTheITGuy** [comment]
Note that I have created a fresh PR for this, as I merged things I shouldn't have mid PR..  Please continue the conversation over there

https://github.com/adafruit/circuitpython/pull/9149

**Comment by RetiredWizard** [comment]
> I tested display, wifi, web workflow, and sd card. All looks good to me.

Nice! If you have the capacitive touch, could you load these two files and try running the GT911Demo.py program?
https://github.com/RetiredWizard/PyDOS_virtkeyboard/blob/main/GT911demo.py
https://github.com/RetiredWizard/PyDOS_virtkeyboard/blob/main/lib/gt911_touch.py

**Comment by SeanTheITGuy** [comment]
Those two file are how I know it worked.
[{'id': 0, 'y': 320, 'x': 502}]
[{'id': 0, 'y': 299, 'x': 326}]
[{'id': 0, 'y': 89, 'x': 446}]
[{'id': 0, 'y': 235, 'x': 245}]

**Pull request closed without merging**